scripts/build-newsroom-videos.js

```diff
@@ -13,32 +12,37 @@ async function buildNewsroomVideos() {
             type: 'video',
             order: 'Date',
             maxResults: 5,
-        }))
-        data = await response.json()
-        const videoDataItems = data.items.map((video) => {
-            return {
-                image_url: video.snippet.thumbnails.high.url,
-                title: video.snippet.title,
-                description: video.snippet.description,
-                videoId: video.id.videoId,
-            }
-        })
-        const videoData = JSON.stringify(videoDataItems, null, '  ');
-        console.log('The following are the Newsroom Youtube videos: ', videoData)
-
-        try {
-            writeFileSync(
-                resolve(__dirname, '../config', 'newsroom_videos.json'),
-                videoData
-            );
-        } catch (err) {
-            console.error(err);
+        }));
+
+        if (!response.ok) {
+            throw new Error(`HTTP error! with status code: ${response.status}`);
+        }
+
+        const data = await response.json();
+
+        if (!data || !data.items) {
+            throw new Error('Invalid data structure received from YouTube API');
         }
+
+        const videoDataItems = data.items.map((video) => ({
+            image_url: video.snippet.thumbnails.high.url,
+            title: video.snippet.title,
+            description: video.snippet.description,
+            videoId: video.id.videoId,
+        }));
+
+        const videoData = JSON.stringify(videoDataItems, null, '  ');
+        console.log('The following are the Newsroom Youtube videos: ', videoData);
+
+        writeFileSync(writePath, videoData);
+
         return videoData;
     } catch (err) {
-        console.log(err)
+        throw new Error(`Failed to build newsroom videos: ${err.message}`);
     }
 }
 
-buildNewsroomVideos()
-module.exports={buildNewsroomVideos}
\ No newline at end of file
+buildNewsroomVideos(resolve(__dirname, '../config', 'newsroom_videos.json'))
+    .catch(err => console.error(err));
```

Comment:
Why have you introduced this line? We shouldn't be catching errors while calling a function initially